frontend/src/routes/train.jsx

```diff
@@ -0,0 +1,116 @@
+import { styled } from "solid-styled-components";
+import { createFormGroup, createFormControl } from "solid-forms";
+import Button from "../components/form/button";
+import EmailInput from "../components/form/email-input";
+import HashInput from "../components/form/hash-input";
+import FileInput from "../components/form/file-input";
+import Slider from "../components/slider";
+import { getFilesUrls, validateEmail } from "../helpers";
+import { postTrain } from "../api";
+import { config } from "../config";
+
+const Train = () => {
+  const group = createFormGroup({
+    email: createFormControl("", {
+      validators: (value) =>
+        validateEmail(value) ? null : { errorMessage: "Invalid email address" },
+    }),
+    files: createFormControl([], {
+      validators: (files) =>
+        files.length < config.minFiles
+          ? { errorMessage: `Number files must be ${config.minFiles} or more` }
+          : null,
+    }),
+    hash: createFormControl("", { readonly: true, disabled: true }),
+  });
+
+  const imageList = () => getFilesUrls(group.controls.files.value);
```

Comment:
Store the `imageList` in the `localStorage` to preserve images between page updates

Reply:
example of [indexedDB](https://gist.github.com/JamesMessinger/a0d6389a5d0e3a24814b)

Reply:
Examples I found: [1](https://hacks.mozilla.org/2012/02/storing-images-and-files-in-indexeddb/), [2](https://hacks.mozilla.org/2012/02/saving-images-and-files-in-localstorage/), [3](https://web.dev/indexeddb-best-practices/), [4](https://codepen.io/steelz/pen/QWLXRMB)